workerapi/server/server.go

```diff
@@ -35,19 +35,25 @@ func MakeServer(
 // (avoid conflict with other injected integers)
 type StatsCollectInterval time.Duration
 
+// a func that inspects the machine this worker is running on and returns
+// an identifying int64 for its storage drive type. It is assumed that
+// 0 corresponds to an SSD & 1 to an HDD
+type DriveTypeFunc func() int64
```

Comment:
Only suggestion is returning an iota int64 type with an entry for unknown that can be returned e.g. when the func is not compatible with the os